include/smack/SplitStructLoadStore.h

```diff
@@ -0,0 +1,31 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+
+#include "llvm/IR/DataLayout.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+#include "llvm/IR/IRBuilder.h"
+
+#include <vector>
+#include <utility>
+
+namespace smack {
+
+class SplitStructLoadStore : public llvm::ModulePass {
```

Comment:
According to the [LLVM docs](http://llvm.org/docs/WritingAnLLVMPass.html#pass-classes-and-requirements) it seems like this should be a `BasicBlockPass` and not a `ModulePass`:

> When choosing a superclass for your Pass, you should choose the most specific class possible, while still being able to meet the requirements listed. This gives the LLVM Pass Infrastructure information necessary to optimize how passes are run, so that the resultant compiler isn’t unnecessarily slow.